processing/src/main/java/org/apache/druid/query/DataSource.java

```diff
@@ -118,6 +123,27 @@ public interface DataSource
    */
   DataSource withUpdatedDataSource(DataSource newSource);
 
+  default DataSource mapWithRestriction(Map<String, Optional<DimFilter>> rowFilters)
+  {
+    return mapWithRestriction(rowFilters, true);
+  }
+
+  /**
+   * Returns an updated datasource based on the policy restrictions on tables. If this datasource contains no table, no
+   * changes should occur.
+   *
+   * @param rowFilters a mapping of table names to row filters, every table in the datasource tree must have an entry
+   * @return the updated datasource, with restrictions applied in the datasource tree
+   */
+  default DataSource mapWithRestriction(Map<String, Optional<DimFilter>> rowFilters, boolean enableStrictPolicyCheck)
```

Comment:
> To allow for the druid_internal or admin case, we can bypass the strict check if the user has permission for STATE READ (a broad administrative permission).

Actually upon further reflection this seems too complex. We don't want to have to consider both policies and STATE permissions. Instead, let's introduce a `Policy` that is of type `admin`. It doesn't apply any restrictions, but it's something an authorizer can return to signify that the user is OK to query unrestricted.

Btw, the strict check in `QueryLifecycle` would need to happen even if the authorized returns `ALLOW`. (Strict check should fail in this case)

Reply:
yes strict check should happen. btw, i added this flag because some tests in views are failing, which i assume won't be an easy solution to fix.

in theory if there's no views, we could default to strict check, it just wants to see table has an entry in policyMap (could Optional.empty() if authorizer says there's no policy).

Reply:
Ah, I see. The strict check should be even stricter: there should be a mode that requires all authorization results to have some non-empty set of policies. The idea with that check is it's a defense against the authorizer being mis-configured in such a way that policies aren't being reported properly.

Reply:
Ideally if we have a PolicyConfig class similar to AuthConfig, or put it in a policy context, maybe could be more flexible.